File: ios/Sources/CrossPromoSDK/Models/PromoApp.swift
import Foundation

public struct PromoApp: Codable, Identifiable {
    public var id: String { appId }

    public let appId: String
    public let appName: String
    public let shortDescription: String
    public let iconUrl: String
    public let platforms: [String]
    public let storeLinks: [String: String]
    public let iosAppStoreId: String
    public let priority: Int
    public let enabled: Bool

    public init(
        appId: String = "",
        appName: String = "",
        shortDescription: String = "",
        iconUrl: String = "",
        platforms: [String] = [],
        storeLinks: [String: String] = [:],
        iosAppStoreId: String = "",
        priority: Int = 10,
        enabled: Bool = true
    ) {
        self.appId = appId
        self.appName = appName
        self.shortDescription = shortDescription
        self.iconUrl = iconUrl
        self.platforms = platforms
        self.storeLinks = storeLinks
        self.iosAppStoreId = iosAppStoreId
        self.priority = priority
        self.enabled = enabled
    }
}

File: android/cross-promo/src/main/kotlin/kr/dev/hoony/crosspromo/model/PromoApp.kt
package kr.dev.hoony.crosspromo.model

data class PromoApp(
    val appId: String = "",
    val appName: String = "",
    val shortDescription: String = "",
    val iconUrl: String = "",
    val platforms: List<String> = emptyList(),
    val storeLinks: Map<String, String> = emptyMap(),
    val iosAppStoreId: String = "",
    val priority: Int = 10,
    val enabled: Boolean = true,
)
